modules/salt/pages/custom-states.adoc

```diff
@@ -0,0 +1,151 @@
+. Click btn:[Update Channel] to save your state.
+
+
+
+.Procedure: Adding Additional Files to a Custom State Channel
+. In the {productname} {webui}, navigate to menu:Configuration[Channels].
+. Click the name of the channel you want to add files to.
+. To create a new file, click btn:``Create configuration file`` and type the contents of the file.
+. To upload an existing file, click btn:``Upload Configuration Files`` and select the file to upload.
+. To copy an existing file, click btn:``Import a File from Another Channel or System`` and select the file to copy.
+
+
+
+.Procedure: Editing a Custom Salt State
+. In the {productname} {webui}, navigate to menu:Configuration[Channels].
+. Click btn:[View/Edit <filename>.sls File].
+. Make your changes to the file
+. Click btn:[Update Configuration File] to save your state.
+
+You can also manage revisions, compare the state to others in your organization, and download the ``.sls`` file from this dialog.
+
+
+
+.Procedure: Assigning a Client to a Custom Salt State
+. In the {productname} {webui}, navigate to menu:Configuration[Channels].
+. Click the name of the state you want to assign a client to.
+. Navigate to the menu:Systems[Target Systems] tab.
+. Check the clients you want to assign.
+. Click btn:[Subscribe systems].
+
+
+
+== Example Custom State Files
+
+This section contains some example custom state files.
+Use these as a basis for writing your own custom states.
+
+
+
+.Example: Initial State File
+
+----
+file.managed:
+  - source: salt://example_org/example_channel/etc/motd
+[
+  FIXME: reference other file here somehow
+]
+----
+
+
+
+.Example: Custom State File
+
+----
+install required packages:
+  pkg.installed:
+    - pkgs:
+      - lsof
+    - failhard: True
+update packages:
+  module.run:
+    - name: pkg.upgrade
+    - failhard: True
+check if reboot is needed:
+  salt.set_reboot_needed:
+    - failhard: True
+reboot:
+   salt.reboot_if_needed:
+     - failhard: True
+----
+
+
+
+== Custom State to Trust a GPG Key
+
+By default, operating systems trust only their own GPG keys when they are installed, and do not trust keys provided by third party packages.
+The clients can be successfully bootstrapped without the GPG key being trusted.
+However, you cannot install new client tool packages or update them until the keys are trusted.
+
+Salt clients are set to trust {suse} tools channels GPG keys when they are bootstrapped.
+For all other clients and channels, you need to manually trust third party GPG keys.
+
+If you are bootstrapping Salt clients from the {productname} {webui}, you can use a custom Salt state to trust the GPG key.
+
+
+
+.Procedure: Trusting a GPG Key With a Custom Salt State
+. On the {productname} Server, at the command prompt, check the contents of the [path]``/srv/www/htdocs/pub/`` directory.
+  This directory contains all available public keys.
+  Take a note of the key that applies to the channel assigned to the client you are registering.
```

Comment:
This directory contains keys we ship, but in most cases these are not the keys the customer has to trust manually.
We should concentrate on cases where a complete new key is needed.

1. find the key which needs to be trusted. Check the web page of the vendor or ask a key server when the keyid is known. Check the fingerprint to verify, that the correct key is used.
2. store the key at a place where client can download it. For example `/srv/www/htdocs/pub/` where we also store our keys

... continued below